### ISSUE TITLE
normalize_homography returns nans on cuda on windows 10

### ISSUE THREAD

**Issue by vishwa91** [kornia/kornia#888]
## 🐛 Bug

I am replicating `kornia.warp_affine` function to get just the coordinates. The function `kornia.geometry.warp.normalize_homography` returns `nan` in some of the entries in the homography matrix on a Windows 10 system, but not on Linux system. I think this is a CUDA synchronization issue, and I have stated my reason in the expected behavior subsection.

## To Reproduce

1. Please see the code below.

```python

import numpy as np

import torch
from torch import nn
import torch.nn.functional as F

import kornia
from kornia.geometry.conversions import deg2rad
from kornia.geometry.conversions import normalize_pixel_coordinates
from kornia.geometry.conversions import convert_affinematrix_to_homography
from kornia.geometry.warp import normalize_homography
from kornia.geometry.transform.imgwarp import get_rotation_matrix2d

if __name__ == '__main__':
    M = torch.zeros(2, 2, 3, device='cuda')
    src_size = (256, 256)
    dst_size=(256, 256)
    align_corners = False

    # Define a basic M with no translation or rotation
    M[:, 0, 0] = 1
    M[:, 1, 1] = 1
    
    B = M.shape[0]

    # This step does not give nans
    M_3x3 = convert_affinematrix_to_homography(M)

    # This step does
    dst_norm_trans_src_norm = normalize_homography(
        M_3x3, src_size, dst_size)

    # Output of dst_norm_trans_src_norm on Windows 10 system
    # tensor([[[nan, nan, nan],
    #              [0., 1., 0.],
    #              [0., 0., 1.]],
    #
    #              [[nan, nan, nan],
    #              [0., 1., 0.],
    #              [0., 0., 1.]]], device='cuda:0')

    # So do consecutive steps
    src_norm_trans_dst_norm = torch.inverse(dst_norm_trans_src_norm)
    coords = F.affine_grid(src_norm_trans_dst_norm[:, :2, :],
                           [B, 1, dst_size[0], dst_size[1]],
                           align_corners=align_corners)

```

## Expected behavior

Under either of the two conditions the matrix prints the expected output:
```python
tensor([[[1., 0., 0.],
             [0., 1., 0.],
             [0., 0., 1.]],
    
            [[1., 0., 0.],
             [0., 1., 0.],
             [0., 0., 1.]]], device='cuda:0')
```
Condition 1: I run this code on a linux machine
Condition 2: I have a either a `torch.cuda.synchronize()` or print statement just after `dst_norm_trans_src_norm = normalize_homography(...)`

## Environment

```
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: (x86_64-posix-seh, Built by strawberryperl.com project) 8.3.0

Clang version: Could not collect
CMake version: version 3.18.2

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2080 SUPER
Nvidia driver version: 460.20
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2\bin\cudnn_ops_train64_8.dll
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.0
[pip3] numpydoc==1.1.0
[pip3] pytorch-model-summary==0.1.2
[pip3] pytorch-wavelets==1.3.0
[pip3] torch==1.7.1
[pip3] torch-dct==0.1.5
[pip3] torchaudio==0.7.2
[pip3] torchvision==0.8.2
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.2.89              h74a9793_1  
[conda] mkl                       2020.1                      216  
[conda] mkl-service               2.3.0            py38hb782905_0  
[conda] mkl_fft                   1.1.0            py38h45dec08_0  
[conda] mkl_random                1.1.1            py38h47e9c7a_0  
[conda] numpy                     1.20.0                   pypi_0    pypi
[conda] numpydoc                  1.1.0                      py_0  
[conda] pytorch                   1.7.1           py3.8_cuda102_cudnn7_0    pytorch
[conda] pytorch-model-summary     0.1.2                    pypi_0    pypi
[conda] pytorch-wavelets          1.3.0                    pypi_0    pypi
[conda] torch-dct                 0.1.5                    pypi_0    pypi
[conda] torchaudio                0.7.2                      py38    pytorch
[conda] torchvision               0.8.2                py38_cu102    pytorch

```

**Comment by edgarriba** [comment]
@vishwa91 thanks for reporting this. My guess would be that `torch.inverse` in https://github.com/kornia/kornia/blob/master/kornia/geometry/warp/homography_warper.py#L366 produce an unexpected behavior with Windows. At this moment I do not have a machine with windows to test. I would suggest to take the code from [`normalize_homography`](https://github.com/kornia/kornia/blob/master/kornia/geometry/warp/homography_warper.py#L340-L374) and inspect what operator return the nan values.

**Comment by vishwa91** [comment]
@edgarriba thank you for the response. I think `torch.inverse` is the culprit. I was able to solve the problem by setting `CUDA_LAUNCH_BLOCKING` to 1, as suggested here https://github.com/pytorch/pytorch/issues/47272#issuecomment-7216123235

**Comment by edgarriba** [comment]
@vishwa91 can we close this since seems to be a pytorch side bug ?

**Comment by vishwa91** [comment]
@edgarriba, yes we should close this issue. Thanks for the reminder!
